File: src/main/java/kr/co/lunatalk/domain/order/dto/response/OrderItemDto.java
package kr.co.lunatalk.domain.order.dto.response;


import kr.co.lunatalk.domain.order.domain.OrderItem;

public record OrderItemDto(
	Long productId,
	String productName,
	Long price,
	Integer quantity,
	Long totalPrice,
	String color
) {
	public static OrderItemDto from(OrderItem orderItem) {
		return new OrderItemDto(
			orderItem.getProductId(),
			orderItem.getProductName(),
			orderItem.getPrice(),
			orderItem.getQuantity(),
			orderItem.getTotalPrice(),
			orderItem.getOptionSnapshot().getColor()
		);
	}
}

File: src/main/kotlin/kr/co/lunatalk/domain/order/dto/response/OrderItemDto.kt
package kr.co.lunatalk.domain.order.dto.response

import kr.co.lunatalk.domain.order.domain.OrderItem

data class OrderItemDto(
    val productId: Long?,
    val productName: String?,
    val price: Long?,
    val quantity: Int?,
    val totalPrice: Long?,
    val color: String?
) {
    companion object {
        fun from(orderItem: OrderItem): OrderItemDto {
            return OrderItemDto(
                productId = orderItem.productId,
                productName = orderItem.productName,
                price = orderItem.price,
                quantity = orderItem.quantity,
                totalPrice = orderItem.totalPrice,
                color = orderItem.optionSnapshot?.color
            )
        }
    }
}
